manifest/oci.go

```diff
@@ -241,3 +251,34 @@ func (m *OCI1) ImageID([]digest.Digest) (string, error) {
 	}
 	return m.Config.Digest.Hex(), nil
 }
+
+// IsEncryptedLayer indicates whether the blob is is encrypted
+func IsEncryptedLayer(b types.BlobInfo) bool {
+	// Check if encrypted mediatype
+	return b.MediaType == ociencspec.MediaTypeLayerGzipEnc ||
+		b.MediaType == ociencspec.MediaTypeLayerEnc
+}
+
+// GetEncryptedMediaType will return the mediatype to its encrypted counterpart and return
+// an error if the mediatype does not support encryption
+func GetEncryptedMediaType(mediatype string) (string, error) {
+	switch mediatype {
+	case DockerV2Schema2LayerMediaType, imgspecv1.MediaTypeImageLayerGzip:
+		return ociencspec.MediaTypeLayerGzipEnc, nil
+	case imgspecv1.MediaTypeImageLayer:
+		return ociencspec.MediaTypeLayerEnc, nil
+	}
+	return "", errors.Errorf("unsupported mediatype to encrypt: %v", mediatype)
+}
+
+// GetDecryptedMediaType will return the mediatype to its decrypted counterpart and return
+// an error if the mediatype does not support encryption
+func GetDecryptedMediaType(mediatype string) (string, error) {
```

Comment:
I’d much prefer for this not to be a public part of the stable API. Is there even anything that calls it?